doc/py_tutorials/py_setup/py_setup_in_ubuntu/py_setup_in_ubuntu.markdown

```diff
@@ -0,0 +1,242 @@
+you want to enable it, you need to install TBB first. ( Also while configuring installation with
+CMake, don't forget to pass -D WITH_TBB=ON. More details below.)
+<br>
+` sudo apt-get install tbb-devel`<br>
+### Downloading OpenCV
+
+Next we have to download OpenCV. You can download the latest release of OpenCV from [sourceforge
+site](http://sourceforge.net/projects/opencvlibrary/). Then extract the folder.
+
+Or you can download latest source from OpenCV's github repo. (If you want to contribute to OpenCV,
+choose this. It always keeps your OpenCV up-to-date). For that, you need to install **Git** first.
+<br>
+`sudo apt-get install git`<br>
+`git clone https://github.com/opencv/opencv.git`
+<br>
+It will create a folder OpenCV in home directory (or the directory you specify). The cloning may
+take some time depending upon your internet connection.
+
+Now open a terminal window and navigate to the downloaded OpenCV folder. Create a new build folder
+and navigate to it.
+<br>
+`mkdir build`<br>
+`cd build`
+<br>
+### Configuring and Installing
+
+Now we have installed all the required dependencies, let's install OpenCV. Installation has to be
+configured with CMake. It specifies which modules are to be installed, installation path, which
+additional libraries to be used, whether documentation and examples to be compiled etc. Below
+command is normally used for configuration (executed from build folder).
+<br>
+`cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local ..`
+<br>
+It specifies that build type is "Release Mode" and installation path is /usr/local. Observe the -D
+before each option and .. at the end. In short, this is the format:
+<br>
+cmake [-D <flag>] [-D <flag>] ..
+<br>
+You can specify as many flags you want, but each flag should be preceded by -D.<br>
+
+So in this tutorial, we are installing OpenCV with TBB. We also build the
+documentation, but we exclude Performance tests and building samples. We also disable GPU related
+modules (since we use OpenCV-Python, we don't need GPU related modules. It saves us some time).<br>
+
+*(All the below commands can be done in a single cmake statement, but it is split here for better
+understanding.)*
+
+-   Enable TBB and Eigen support:
+    <br>
+    `cmake -D WITH_TBB=ON -D WITH_EIGEN=ON ..`
+    <br>
+-   Enable documentation and disable tests and samples<br>
+    
+    `cmake -D BUILD_DOCS=ON -D BUILD_TESTS=OFF -D BUILD_PERF_TESTS=OFF -D BUILD_EXAMPLES=OFF ..`<br>
+   <br>
+-   Disable all GPU related modules.<br>
+    <br>
+    `cmake -D WITH_OPENCL=OFF -D WITH_CUDA=OFF -D BUILD_opencv_gpu=OFF -D BUILD_opencv_gpuarithm=OFF -D BUILD_opencv_gpubgsegm=OFF -D BUILD_opencv_gpucodec=OFF -D BUILD_opencv_gpufeatures2d=OFF -D BUILD_opencv_gpufilters=OFF -D BUILD_opencv_gpuimgproc=OFF -D BUILD_opencv_gpulegacy=OFF -D BUILD_opencv_gpuoptflow=OFF -D BUILD_opencv_gpustereo=OFF -D BUILD_opencv_gpuwarping=OFF ..`
```

Comment:
This list is wrong. `-DWITH_CUDA=OFF` is enough. CUDA support is disabled if there is no CUDA SDK detected.

Why OpenCL is disabled? (it doesn't reduce compilation time significantly)